packages/matter.js/src/protocol/interaction/InteractionServer.ts

```diff
@@ -371,17 +388,248 @@ export function attributePathToId({ endpointId, clusterId, attributeId }: TypeFr
+        }
+        const attribute = this.attributes.get(attributePathToId({ endpointId, clusterId, attributeId }));
+        const attributeName = `${attribute?.name ?? "unknown"}(${this.toHex(attributeId)})`;
+        return `${endpointName}/${clusterName}/${attributeName}`;
+    }
+
+    resolveEventName({ endpointId, clusterId, eventId }: TypeFromSchema<typeof TlvEventPath>) {
+        if (endpointId === undefined) {
+            return `*/${this.toHex(clusterId)}/${this.toHex(eventId)}`;
+        }
+        const endpoint = this.endpoints.get(endpointId);
+        if (endpoint === undefined) {
+            return `unknown(${this.toHex(endpointId)})/${this.toHex(clusterId)}/${this.toHex(eventId)}`;
+        }
+        const endpointName = `${endpoint.name}(${this.toHex(endpointId)})`;
+
+        if (clusterId === undefined) {
+            return `${endpointName}/*/${this.toHex(eventId)}`;
+        }
+        const cluster = endpoint.getClusterServerById(clusterId);
+        if (cluster === undefined) {
+            return `${endpointName}/unknown(${this.toHex(clusterId)})/${this.toHex(eventId)}`;
+        }
+        const clusterName = `${cluster.name}(${this.toHex(clusterId)})`;
+
+        if (eventId === undefined) {
+            return `${endpointName}/${clusterName}/*`;
+        }
+        const event = this.events.get(eventPathToId({ endpointId, clusterId, eventId }));
+        const eventName = `${event?.name ?? "unknown"}(${this.toHex(eventId)})`;
+        return `${endpointName}/${clusterName}/${eventName}`;
+    }
+
+    resolveCommandName({ endpointId, clusterId, commandId }: TypeFromSchema<typeof TlvCommandPath>) {
+        if (endpointId === undefined) {
+            return `*/${this.toHex(clusterId)}/${this.toHex(commandId)}`;
+        }
+        const endpoint = this.endpoints.get(endpointId);
+        if (endpoint === undefined) {
+            return `unknown(${this.toHex(endpointId)})/${this.toHex(clusterId)}/${this.toHex(commandId)}`;
+        }
+        const endpointName = `${endpoint.name}(${this.toHex(endpointId)})`;
+
+        if (clusterId === undefined) {
+            return `${endpointName}/*/${this.toHex(commandId)}`;
+        }
+        const cluster = endpoint.getClusterServerById(clusterId);
+        if (cluster === undefined) {
+            return `${endpointName}/unknown(${this.toHex(clusterId)})/${this.toHex(commandId)}`;
+        }
+        const clusterName = `${cluster.name}(${this.toHex(clusterId)})`;
+
+        if (commandId === undefined) {
+            return `${endpointName}/${clusterName}/*`;
+        }
+        const command = this.commands.get(commandPathToId({ endpointId, clusterId, commandId }));
+        const attributeName = `${command?.name ?? "unknown"}(${this.toHex(commandId)})`;
+        return `${endpointName}/${clusterName}/${attributeName}`;
+    }
+
+    getAttributes(filters: TypeFromSchema<typeof TlvAttributePath>[], onlyWritable = false): AttributeWithPath[] {
```

Comment:
Similar comment to above although I know it's a bit more tricky with the onlyWritable specialization for attributes.

Reply:
exactly thi was my main topic ... but we could post process. WIll think

Reply:
But here it is more hard and bgger changes and typing issues ... owuld leav" slighly duplicated code" for now